Firebase.h

```diff
@@ -25,49 +25,97 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
- public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
-  }
- private:
-  int _code = 0;
-  String _message = "";
-};
+//TODO(edcoyne) split these into multiple files.
+
+class FirebaseCall;
+class FirebaseEventStream;
 
-// Firebase is the connection to firebase.
+// Primary client to the Firebase backend.
 class Firebase {
  public:
   Firebase(const String& host);
   Firebase& auth(const String& auth);
-  const FirebaseError& error() const {
-    return _error;
+
+  // Fetch result at "path" to a local variable. If the value is too large you will exceed
+  // local memory.
+  FirebaseCall get(const String& path);
```

Comment:
I think there are a couple of issues with holding it even assuming non-concurrency. One is memory usage, on a memory constrained device having potentially large chunks of memory sitting unused can be an issue. The other is if I remember TCP/HTTP correctly than on a large request part of this will actually be held on the server so this could have larger effects. (i.e. if memory serves, i should verify, but if the client buffer is full the server will buffer the rest until the client resumes reading.